executor/common_linux.h

```diff
@@ -1570,6 +1570,310 @@ static long syz_init_net_socket(volatile long domain, volatile long type, volati
+	uint16 handle;
+	bdaddr_t bdaddr;
+	uint8 link_type;
+	uint8 encr_mode;
+} __attribute__((packed));
+
+#define HCI_EV_CONN_REQUEST 0x04
+struct hci_ev_conn_request {
+	bdaddr_t bdaddr;
+	uint8 dev_class[3];
+	uint8 link_type;
+} __attribute__((packed));
+
+#define HCI_EV_REMOTE_FEATURES 0x0b
+struct hci_ev_remote_features {
+	uint8 status;
+	uint16 handle;
+	uint8 features[8];
+} __attribute__((packed));
+
+#define HCI_EV_CMD_COMPLETE 0x0e
+struct hci_ev_cmd_complete {
+	uint8 ncmd;
+	uint16 opcode;
+} __attribute__((packed));
+
+#define HCI_OP_READ_BUFFER_SIZE 0x1005
+struct hci_rp_read_buffer_size {
+	uint8 status;
+	uint16 acl_mtu;
+	uint8 sco_mtu;
+	uint16 acl_max_pkt;
+	uint16 sco_max_pkt;
+} __attribute__((packed));
+
+#define HCI_OP_READ_BD_ADDR 0x1009
+struct hci_rp_read_bd_addr {
+	uint8 status;
+	bdaddr_t bdaddr;
+} __attribute__((packed));
+
+struct hci_dev_stats {
+	uint32 err_rx;
+	uint32 err_tx;
+	uint32 cmd_tx;
+	uint32 evt_rx;
+	uint32 acl_tx;
+	uint32 acl_rx;
+	uint32 sco_tx;
+	uint32 sco_rx;
+	uint32 byte_rx;
+	uint32 byte_tx;
+};
+
+struct hci_dev_info {
+	uint16 dev_id;
+	char name[8];
+
+	bdaddr_t bdaddr;
+
+	uint32 flags;
+	uint8 type;
+
+	uint8 features[8];
+
+	uint32 pkt_type;
+	uint32 link_policy;
+	uint32 link_mode;
+
+	uint16 acl_mtu;
+	uint16 acl_pkts;
+	uint16 sco_mtu;
+	uint16 sco_pkts;
+
+	struct hci_dev_stats stat;
+};
+
+struct hci_dev_req {
+	uint16 dev_id;
+	uint32 dev_opt;
+};
+
+#define HCISETSCAN _IOW('H', 208, int)
+#define HCIGETDEVINFO _IOR('H', 211, int)
+
+#define BUF_SIZE 4096
+#define ACL_HANDLE 256
+#define HCI_DEV 0
+
+static int vhci_fd = -1;
+static int vhci_init = 0;
+
+static long syz_init_net_socket(volatile long domain, volatile long type, volatile long proto);
+
+static int hci_send_event_packet(int fd, uint8 evt, void* data, size_t data_len)
+{
+	struct iovec iv[3];
+
+	struct hci_event_hdr hdr;
+	hdr.evt = evt;
+	hdr.plen = data_len;
+
+	uint8 type = HCI_EVENT_PKT;
+
+	iv[0].iov_base = &type;
+	iv[0].iov_len = sizeof(type);
+	iv[1].iov_base = &hdr;
+	iv[1].iov_len = sizeof(hdr);
+	iv[2].iov_base = data;
+	iv[2].iov_len = data_len;
+
+	return writev(fd, iv, sizeof(iv) / sizeof(struct iovec));
+}
+
+static int hci_send_event_cmd_complete(int fd, uint16 opcode, void* data, size_t data_len)
+{
+	struct iovec iv[4];
+
+	struct hci_event_hdr hdr;
+	hdr.evt = HCI_EV_CMD_COMPLETE;
+	hdr.plen = sizeof(struct hci_ev_cmd_complete) + data_len;
+
+	struct hci_ev_cmd_complete evt_hdr;
+	evt_hdr.ncmd = 1;
+	evt_hdr.opcode = opcode;
+
+	uint8 type = HCI_EVENT_PKT;
+
+	iv[0].iov_base = &type;
+	iv[0].iov_len = sizeof(type);
+	iv[1].iov_base = &hdr;
+	iv[1].iov_len = sizeof(hdr);
+	iv[2].iov_base = &evt_hdr;
+	iv[2].iov_len = sizeof(evt_hdr);
+	iv[3].iov_base = data;
+	iv[3].iov_len = data_len;
+
+	return writev(fd, iv, sizeof(iv) / sizeof(struct iovec));
+}
+
+static void process_command_pkt(int fd, char* buf, size_t buf_size)
+{
+	struct hci_command_hdr* hdr = (struct hci_command_hdr*)(buf + 1);
```

Comment:
If you add uint8 type as first member into hci_command_hdr, it will simplify code and avoid casts.

Reply:
Ditto.